src/main/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaProducer.java

```diff
@@ -0,0 +1,56 @@
+package uk.gov.companieshouse.itemhandler.kafka;
+
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.kafka.exceptions.ProducerConfigException;
+import uk.gov.companieshouse.kafka.message.Message;
+import uk.gov.companieshouse.kafka.producer.Acks;
+import uk.gov.companieshouse.kafka.producer.CHKafkaProducer;
+import uk.gov.companieshouse.kafka.producer.ProducerConfig;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+import static uk.gov.companieshouse.itemhandler.ItemHandlerApplication.APPLICATION_NAMESPACE;
+
+@Service
+public class OrdersKafkaProducer implements InitializingBean {
+    private static final Logger LOGGER = LoggerFactory.getLogger(APPLICATION_NAMESPACE);
+    private CHKafkaProducer chKafkaProducer;
+    @Value("${spring.kafka.consumer.bootstrap-servers}")
+    private String brokerAddresses;
+
+    /**
+     * Sends message to Kafka topic
+     * @param message message
+     * @throws ExecutionException
+     * @throws InterruptedException
+     */
+    public void sendMessage(final Message message) throws ExecutionException, InterruptedException {
+        LOGGER.trace("Sending message to kafka topic.");
+        chKafkaProducer.send(message);
+    }
+
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        LOGGER.trace("Configuring CH Kafka producer");
+        ProducerConfig config = new ProducerConfig();
+        if (brokerAddresses != null && !brokerAddresses.isEmpty()) {
+            config.setBrokerAddresses(brokerAddresses.split(","));
+        } else {
+            throw new ProducerConfigException("Broker addresses for kafka broker missing, check if environment variable KAFKA_BROKER_ADDR is configured. " +
+                    "[Hint: The property 'kafka.broker.addresses' uses the value of this environment variable in live environments " +
+                    "and that of 'spring.embedded.kafka.brokers' property in test.]");
+        }
+
+        config.setRequestTimeoutMilliseconds(3000);
+        config.setRoundRobinPartitioner(true);
+        config.setAcks(Acks.WAIT_FOR_ALL);
+        config.setRetries(10);
+        chKafkaProducer = new CHKafkaProducer(config);
+    }
+}
```

Comment:
Why does this PR include a Kafka producer if it is concerned with resilient consumption? Is this how messages are shunted from topic to topic?

Reply:
Yes. When a consumer fails to process a message (even with retryable error) the message must be published to the next topic for recovery. This is when this producer is needed.